src/main/java/org/opensearch/security/user/ThreadContextUserInfo.java

```diff
@@ -0,0 +1,101 @@
+import org.opensearch.security.privileges.TenantPrivileges;
+import org.opensearch.security.support.Base64Helper;
+import org.opensearch.security.support.SecuritySettings;
+
+import static org.opensearch.security.support.ConfigConstants.OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT;
+import static org.opensearch.security.support.ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED;
+import static org.opensearch.security.support.ConfigConstants.USER_ATTRIBUTE_SERIALIZATION_ENABLED_DEFAULT;
+import static org.opensearch.security.support.SecurityUtils.escapePipe;
+
+/**
+ * Functionality to add parseable information about the current user to the thread context. Usually called
+ * in the SecurityFilter.
+ * <p>
+ * Moved from https://github.com/opensearch-project/security/blob/d29095f26dba1a26308c69b608dc926bd40c0f52/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L293
+ */
+public class ThreadContextUserInfo {
+    private static final String READ_ACCESS = "READ";
+    private static final String WRITE_ACCESS = "WRITE";
+    private static final String NO_ACCESS = "NONE";
+    private static final String GLOBAL_TENANT = "global_tenant";
+
+    private volatile boolean userAttributeSerializationEnabled;
+    private final ThreadContext threadContext;
+    private final PrivilegesConfiguration privilegesConfiguration;
+
+    public ThreadContextUserInfo(
+        ThreadContext threadContext,
+        PrivilegesConfiguration privilegesConfiguration,
+        ClusterSettings clusterSettings,
+        Settings settings
+    ) {
+        this.threadContext = threadContext;
+        this.userAttributeSerializationEnabled = settings.getAsBoolean(
+            USER_ATTRIBUTE_SERIALIZATION_ENABLED,
+            USER_ATTRIBUTE_SERIALIZATION_ENABLED_DEFAULT
+        );
+        this.privilegesConfiguration = privilegesConfiguration;
+        clusterSettings.addSettingsUpdateConsumer(
+            SecuritySettings.USER_ATTRIBUTE_SERIALIZATION_ENABLED_SETTING,
+            newIsUserAttributeSerializationEnabled -> {
+                userAttributeSerializationEnabled = newIsUserAttributeSerializationEnabled;
+            }
+        );
+    }
+
+    public void setUserInfoInThreadContext(PrivilegesEvaluationContext context) {
+        if (threadContext.getTransient(OPENDISTRO_SECURITY_USER_INFO_THREAD_CONTEXT) == null) {
+            StringJoiner joiner = new StringJoiner("|");
+            // Escape any pipe characters in the values before joining
+            joiner.add(escapePipe(context.getUser().getName()));
+            joiner.add(escapePipe(String.join(",", context.getUser().getRoles())));
+            joiner.add(escapePipe(String.join(",", context.getMappedRoles())));
+
+            String requestedTenant = context.getUser().getRequestedTenant();
+            joiner.add(requestedTenant);
+
+            String tenantAccessToCheck = getTenancyAccess(context);
+            joiner.add(tenantAccessToCheck);
+
```

Comment:
nit: missing logger from original method- https://github.com/nibix/security/blob/main/src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java#L324

The log line has been helpful to debug configuration/operational issues.